activiti-core/activiti-api-impl/activiti-api-task-runtime-impl/src/main/java/org/activiti/runtime/api/impl/TaskRuntimeImpl.java

```diff
@@ -455,5 +454,38 @@ public void save(SaveTaskPayload saveTaskPayload) {
         }
         throw new IllegalStateException("There is no authenticated user, we need a user authenticated to find tasks");
     }
+    
+    private void assertAssigneeIsACandidateUser(String taskId, String assignee) {
+        List<String> userCandidates = userCandidates(taskId);
+        if(!userCandidates.contains(assignee)){
+            throw new IllegalStateException("You cannot assign a task to " + assignee + " due it is not a candidate for it");
+        }
+    }
+    
+    private void reassignTask(String taskId, String assignee) {
+        releaseTask(taskId);
+        taskService.claim(taskId, assignee);
+    }
+    
+    private void releaseTask(String taskId) {
+        assertCanReleaseTask(taskId);
+        taskService.unclaim(taskId);
+    }
+    
+    private void assertCanReleaseTask(String taskId) {
+        Task task;
+        try {
+            task = task(taskId);
+        } catch (IllegalStateException ex) {
+            throw new IllegalStateException("The authenticated user cannot release task" + taskId + " due it is not a candidate for it");
```

Comment:
I think you are right, but in this case I extracted that method from the release functionality because I needed to release a task before to assign it. I would avoid to change the logic they used in order to be sure that there are no regression in the release functionality.